ui/src/app/views/settings/action/list/action.list.component.ts

```diff
@@ -0,0 +1,40 @@
+import {Component, Input} from '@angular/core';
+import {Action} from '../../../../model/action.model';
+import {Table} from '../../../../shared/table/table';
+import {ActionService} from '../../../../service/action/action.service';
+import {User} from '../../../../model/user.model';
+import {AuthentificationStore} from '../../../../service/auth/authentification.store';
+
+@Component({
+    selector: 'app-action-list',
+    templateUrl: './action.list.html',
+    styleUrls: ['./action.list.scss']
+})
+export class ActionListComponent extends Table {
+    @Input('maxPerPage')
+    set maxPerPage(data: number) {
+        this.nbElementsByPage = data;
+    };
+
+    public ready = true;
+    filter: string;
+    actions: Array<Action>;
+    public currentUser: User;
+
+    constructor(private _actionService: ActionService,
+                private _authentificationStore: AuthentificationStore) {
+        super();
+        this.currentUser = this._authentificationStore.getUser();
+        this._actionService.getActions().subscribe( actions => {
+            this.actions = actions;
+            this.ready = true;
+        });
+    }
+
+    getData(): any[] {
+        if (!this.filter || this.filter === '') {
```

Comment:
when you make `!this.filter` if this.filter is equals to null, undefined, 0 or '' it's true so the part `this.filter === ''` isn't necessary